llvm/test/Transforms/InstCombine/lshr.ll

```diff
@@ -360,8 +360,33 @@ define <3 x i14> @mul_splat_fold_vec(<3 x i14> %x) {
   ret <3 x i14> %t
 }
 
+define i32 @shl_add_lshr(i32 %x, i32 %c, i32 %y) {
```

Comment:
Please rename this test to `shl_add_lshr_flag_preservation`, and add the following basic test:
```
define i32 @shl_add_lshr(i32 %x, i32 %c, i32 %y) {
   %shl = shl nuw i32 %x, %c
   %add = add nuw i32 %shl, %y
   %lshr = lshr i32 %add, %c
   ret i32 %lshr
}
```